openedx/core/djangoapps/safe_endpoints/tests/test_middleware.py

```diff
@@ -0,0 +1,77 @@
+"""
+Unit tests for safe_endpoints Middleware.
+"""
+import ddt
+
+from django.test import TestCase
+from rest_framework.authentication import SessionAuthentication
+from rest_framework_jwt.authentication import JSONWebTokenAuthentication
+from rest_framework.decorators import api_view
+from rest_framework.views import APIView
+
+from openedx.core.djangolib.testing.utils import get_mock_request
+
+from ..middleware import EnsureJWTAuthSettingsMiddleware
+
+
+class MockIncludedPermissionClass(object): pass
+class MockRequiredPermissionClass(object): pass
+
+
+def mock_auth_decorator(include_jwt_auth=True):
+    def _decorator(f):
+        f.permission_classes = (MockIncludedPermissionClass,)
+        f.authentication_classes = (SessionAuthentication,)
+        if include_jwt_auth:
+            f.authentication_classes += (JSONWebTokenAuthentication,)
```

Comment:
I thought you were also going to have something like `MockJwtSubclass` which is based on JSONWebTokenAuthentication?

I know I can be too detailed at times, but maybe this is a case where we want 100% code coverage, given the nature of this.

Reply:
See my comment above.  The `include_jwt_auth` parameter does this.

Reply:
I see.  I was confusing `JSONWebTokenAuthentication` with `BaseJSONWebTokenAuthentication`.  

I think the intention of coverage might be more clear with something like:
```
class SomeJwtAuthenticationSubclass(BaseJSONWebTokenAuthentication): pass
```
But, I now see that the current code does do what I asked as-is.

Reply:
I see.  I was confusing `JSONWebTokenAuthentication` with `BaseJSONWebTokenAuthentication`.  

I think the intention of coverage might be more clear with something like:
```
class SomeJwtAuthenticationSubclass(BaseJSONWebTokenAuthentication): pass
```
But, I now see that the current code does do what I had asked for.